core/token-pooling/redis-token-persistence.js

```diff
@@ -3,13 +3,13 @@
 const { URL } = require('url')
 const Redis = require('ioredis')
 const log = require('../server/log')
-const TokenPersistence = require('./token-persistence')
 
-module.exports = class RedisTokenPersistence extends TokenPersistence {
```

Comment:
At first glance I found myself wondering if it'd be worth trying to keep `TokenPersistence` as a base class or pulling those functions out into a utils that could be composed/injected into a hypothetical future non-redis based persistence implementation.

But after a bit of thinking I really don't think it is. IIRC the `noteToken*` functions are the only consumed/utilized functions, so any hypothetical future non-redis based persistence solution would just need to have those functions anyway like this one

Reply:
Yea, it’s a reasonable question. I think my rationale for removing the abstraction layer is that we don’t need to support more than one way to persist dynamic tokens. It’s only for Shields production.